### PR TITLE
URI schemes are not case-sensitive

### PR DESCRIPTION
**Pull request by butane**
#4758 
_Grepping_ the source revealed a few more similar bugs.

Signed-off-by: Vivek Dinesh <vivekdinesh5@gmail.com>

### CODE REVIEW

File: system/helpers/url_helper.php [hunk 1]
```diff
@@ -443,7 +443,7 @@ function auto_link($str, $type = 'both', $popup = FALSE)
 	 */
 	function prep_url($str = '')
 	{
-		if ($str === 'http://' OR $str === '')
+		if (strtolower($str) === 'http://' OR $str === '')
```
Comment (narfbg) [inline system/helpers/url_helper.php]: This 'http://' check is useless, if anything - it should be removed.
Same thing with the Form validation method.

Comment (butane) [inline system/helpers/url_helper.php]: What is the expected return value for the function call,
prep_url('http://')

Comment (narfbg) [inline system/helpers/url_helper.php]: Don't care for it - it's obviously bad usage.

**Comment by butane** [comment]
Removed the unwanted checks.

**Pull request merged**